### DISCUSSION TITLE
Setting the global skin from lua

### DISCUSSION THREAD

**Discussion by KillTheMule** [SFX-WoW/Masque#369] [Questions] [answered]
Hey, thanks for masque!

I'm looking for the ability to set the global skin  (i.e. the functionality I get from opening the options, choosing the "Global" category and selecting a skin from the dropdown menu) from lua code. Is that exposed somehow? If not, can anyone suggest what to do? I'm kinda pondering mucking with the savedvars and call `Masque:OnEnable()`, but that's of course kinda hacky ...

Thanks for any pointers :)

**Comment by StormFX** [comment]
The root ("Global") group is unavailable externally. All you can do is set the skin for each add-on individually (which will propagate to sub-groups), assuming you know which add-ons are enabled in Masque.

**Reply by KillTheMule** [reply]
Hey thanks for your answer! I'm fine with skinning addons individually. Am I right assuming I need to call `API:GetGroupByID(addonname)` to get the group for the addon? I'm however unclear how to set the skin on that, seeing that the functions used in the config options are not exposed externally. Could you give me a hint on that? Thanks in advance :)

**Reply by StormFX** [reply, marked as answer]
You can use that API function. If that doesn't work, just use the `Group` function and pass the add-on name. To set the skin, use `Group:__Set("SkinID", "NameOfSkin")`.

**Reply by KillTheMule** [reply]
That works nicely, thanks a bunch!
